src/Microsoft.AspNetCore.Mvc.Razor/RazorPageActivator.cs

```diff
@@ -104,6 +108,41 @@ private PageActivationInfo CreateViewActivationInfo(Type type)
             };
         }
 
+        private ViewDataCreateFromSource GetViewDataCreatorFromSource(Type pageType, Type viewDataDictionaryType)
+        {
+            var parameterTypes = new Type[] { typeof(ViewDataDictionary) };
+            var matchingConstructor = viewDataDictionaryType.GetConstructor(parameterTypes);
+            if (matchingConstructor == null)
+            {
+                var message = Resources.FormatViewCannotBeActivated(pageType.FullName, GetType().FullName);
+                throw new InvalidOperationException();
```

Comment:
Why and when would a user ever see this exception? This should be a `Debug.Assert`